pymatgen/io/openmm/sets.py

```diff
@@ -0,0 +1,130 @@
+__email__ = "orion@lbl.gov"
+__date__ = "Nov 2021"
+
+
+class OpenMMSet(InputSet):
+    """
+    An InputSet for OpenMM Simulations.
+
+    This is a container for the serialized topology, system, integrator, and state files.
+    Topology is stored as a pdb and the other files are stored as xml. The state file is
+    optional but positions must be manually assigned if no state is provided.
+    """
+
+    @classmethod
+    def from_directory(
+        cls,
+        directory: Union[str, Path],
+        topology_file: str = "topology.pdb",
+        system_file: str = "system.xml",
+        integrator_file: str = "integrator.xml",
+        state_file: str = "state.xml",
+    ):
+        """
+        Instantiates an InputSet from a directory containing a Topology PDB, a
+        System XML, a Integrator XML, and (optionally) a State XML. If no State
+        is given, system coordinates must be manually assigned.
+
+        Args:
+            directory: directory that holds the input files.
+            topology_file: name of the pdb file with topological information.
+            system_file: name of the serialized System xml file.
+            integrator_file: name of the serialized Integrator xml file.
+            state_file: name of the serialized State xml file. If there is no state_file,
+                then positions must be set for simulation.
+
+        Returns:
+            an OpenMMSet
+        """
+        source_dir = Path(directory)
+        topology_input = TopologyInput.from_file(source_dir / topology_file)
+        system_input = SystemInput.from_file(source_dir / system_file)
+        integrator_input = IntegratorInput.from_file(source_dir / integrator_file)
+        inputs = {
+            topology_file: topology_input,
+            system_file: system_input,
+            integrator_file: integrator_input,
+        }
+        openmm_set = OpenMMSet(
+            inputs=inputs,  # type: ignore
+            topology_file=topology_file,
+            system_file=system_file,
+            integrator_file=integrator_file,
+        )
+        if Path(source_dir / state_file).is_file():
+            openmm_set.inputs[state_file] = StateInput.from_file(source_dir / state_file)
+            openmm_set.state_file = state_file  # should this be a dict-like assignment?
+        return openmm_set
+
+    def validate(self) -> bool:
+        """
+        Checks that the OpenMMSet will generate a valid simulation.
+
+        Returns:
+            bool
+        """
+        try:
+            self.get_simulation()
+        except Exception as e:
+            print(
+                "A valid simulation could not be generated, the following error was raised:",
+                e,
+            )
+            return False
+        else:
+            return True
+
+    def get_simulation(
+        self,
+        platform: Optional[openmm.Platform] = None,
+        platformProperties: Dict[str, str] = None,
+    ) -> Simulation:
```

Comment:
Hey @rkingsbury, I wanted to draw attention to this change and make sure it fits into the new abstraction. 

For context, OpenMM Simulations have a `platform` property that specifies what computational resource to use, e.g. CUDA, OpenCL, or CPU. The platformProperties dictionary specifies things like thread count or GPU device index. In theory, the platform and platform properties should not alter the trajectory, though properties like double vs single precision definitely can.

For the most part, OpenMM will determine the optimal platform and sensible platformProperties. However in certain edge cases, like parallelizing across multiple GPUs on a single node, it is necessary to use these properties.

Since these are platform specific, they don't seem to belong in the input set. Adding them as optional arguments to `get_simulation` was the cleanest solution I could think of here.